device/alpaka/include/traccc/alpaka/clusterization/clusterization_algorithm.hpp

```diff
@@ -57,12 +64,19 @@ class clusterization_algorithm
 
     private:
     /// The average number of cells in each partition
-    using config_type = unsigned short;
-    config_type m_target_cells_per_partition;
+    config_type m_config;
     /// The memory resource(s) to use
     traccc::memory_resource m_mr;
     /// The copy object to use
     std::reference_wrapper<vecmem::copy> m_copy;
+    /// Memory reserved for edge cases
+    vecmem::data::vector_buffer<device::details::index_t> m_f_backup,
+        m_gf_backup;
+    vecmem::data::vector_buffer<unsigned char> m_adjc_backup;
+    vecmem::data::vector_buffer<device::details::index_t> m_adjv_backup;
+    vecmem::unique_alloc_ptr<unsigned int> m_backup_mutex;
+    mutable std::mutex m_setup_mutex;
+    mutable bool m_mutex_is_setup = false;
```

Comment:
The goal here is to do the setup strictly just once, even in a multi-threaded environment, right?

The standard way of doing that is with [std::call_once](https://en.cppreference.com/w/cpp/thread/call_once). I.e. put a mutable [std::once_flag](https://en.cppreference.com/w/cpp/thread/once_flag) member into the class, and then run the necessary kernel inside of an `std::call_once` call. :thinking: (This comes from having run into some issues in Athena when we did this sort of a thing without `std::call_once`...)

Reply:
That's a good idea.